playground/backend/internal/db/datastore/datastore_db.go

```diff
@@ -243,6 +244,223 @@ func (d *Datastore) GetCatalog(ctx context.Context, sdkCatalog []*entity.SDKEnti
 	}), nil
 }
 
+//GetDefaultExamples returns the default examples
+func (d *Datastore) GetDefaultExamples(ctx context.Context, sdks []*entity.SDKEntity) (map[pb.Sdk]*pb.PrecompiledObject, error) {
+	tx, err := d.Client.NewTransaction(ctx, datastore.ReadOnly)
+	if err != nil {
+		logger.Errorf(errorMsgTemplateCreatingTx, err.Error())
+		return nil, err
+	}
+	defer rollback(tx)
+
+	//Retrieving examples
+	var exampleKeys []*datastore.Key
+	for _, sdk := range sdks {
+		exampleKeys = append(exampleKeys, utils.GetExampleKey(sdk.Name, sdk.DefaultExample))
+	}
+	var examplesWithNils = make([]*entity.ExampleEntity, len(exampleKeys))
+	examples := make([]*entity.ExampleEntity, 0)
+	if err = tx.GetMulti(exampleKeys, examplesWithNils); err != nil {
+		if errors, ok := err.(datastore.MultiError); ok {
+			for _, errVal := range errors {
+				if errVal == datastore.ErrNoSuchEntity {
+					for _, exampleVal := range examplesWithNils {
+						if exampleVal != nil {
+							examples = append(examples, exampleVal)
+						}
+					}
+				}
+			}
+		} else {
+			logger.Errorf("error during the getting default examples, err: %s\n", err.Error())
+			return nil, err
+		}
+	} else {
+		examples = examplesWithNils
+	}
+
+	if len(examples) == 0 {
+		logger.Error("no examples")
+		return nil, fmt.Errorf("no examples")
+	}
+
+	//Retrieving snippets
+	var snippetKeys []*datastore.Key
+	for _, exampleKey := range exampleKeys {
+		snippetKeys = append(snippetKeys, utils.GetSnippetKey(exampleKey.Name))
+	}
+	snippetsWithNils := make([]*entity.SnippetEntity, len(snippetKeys))
+	snippets := make([]*entity.SnippetEntity, 0)
+	if err = tx.GetMulti(snippetKeys, snippetsWithNils); err != nil {
```

Comment:
looks like code duplicate, maybe it worth to move it to the separate function?

Reply:
@konstantinurysov i did this way https://github.com/akvelon/beam/pull/217/commits/4064e919f0593545519ed429451eb5c8d1ee3b04

We don't have generic because the project uses 1.16 version :(((, i know that 1.18 has generic types